docs/code/getting-started.md

```diff
@@ -108,53 +108,69 @@ Apps on elementary OS are organized into standardized directories contained in y
 
 Later on, We'll talk about adding other directories like "po" and "data". For now, this is all we need.
 
-## Gtk.Window and Gtk.Button {#gtk-window-and-gtk-button}
+## Gtk.Application {#gtk-application}
 Now what you've been waiting for! We're going to create a window that contains a button. When pressed, the button will display the text "Hello World!" To do this, we're going to use a widget toolkit called GTK+ and the programming language Vala. Before we begin, we highly recommend that you do not copy and paste. Typing each section manually will help you to practice and remember. Let's get started:
 
 1. Create a new file in Scratch and save it as "gtk-hello.vala" inside your "src" folder
 
-2. First we must create the main function of our new GTK app. Type the following in your "gtk-hello.vala".
+2. First we create a special class called a `Gtk.Application`. `Gtk.Application` is a class that handles many important aspects of a Gtk+ app like app uniqueness and the application ID you need to identify your app to the notifications server. If you want some more details about `Gtk.Application`, [check out Valadoc](https://valadoc.org/gtk+-3.0/Gtk.Application). For now, type the following in your "gtk-hello.vala".
+
+        public class MyApp : Gtk.Application {
+
+            public MyApp () {
+                Object (
+                    application_id: "com.github.yourusername.yourrepositoryname",
+                    flags: ApplicationFlags.FLAGS_NONE
+                );
+            }
 
-        int main (string[] args) {
-            Gtk.init (ref args);
+            protected override void activate () {
+                var main_window = new Gtk.ApplicationWindow (this);
+                main_window.show_all ();
+            }
+
+            public static int main (string[] args) {
+                var app = new MyApp ();
+                return app.run (args);
+            }
         }
+        
+    Ready to test it out? Fire up your terminal and make sure you're in "~/Projects/gtk-hello/src". Then execute the following commands to compile and run your first Gtk+ app:
 
-3. Now, that we've initialized Gtk, we'll create a new window and declare a few properties of this window. You'll notice that most of these property names are pretty straightforward. Try to guess what each one does and we'll explain in a second. Type the following after the `Gtk.init` line, but before the last bracket:
+        ```bash
+        valac --pkg gtk+-3.0 gtk-hello.vala
+        ./gtk-hello
+        ```
 
```

Comment:
I got warning "Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -13 and height 17
" and only a small window - is that intended?